src/charm.py

```diff
@@ -100,6 +107,22 @@ def on_config_changed_handler(event, fw_adapter):
 
 
 def on_start_handler(event, fw_adapter):
+    build_pod_spec(fw_adapter)
+
+
+def on_upgrade_handler(event, fw_adapter):
+    on_start_handler(event, fw_adapter)
+
+
+def on_stop_handler(event, fw_adapter):
+    fw_adapter.set_unit_status(MaintenanceStatus("Pod is terminating"))
+
+
+def build_pod_spec(fw_adapter):
```

Comment:
To avoid confusing with `build_juju_pod_spec`, how about we rename this to `set_juju_pod_spec`? Also, that's really what the function is doing too anyway.

Reply:
Great idea! Done.